crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -22,8 +22,13 @@ pub struct EntityRef<'w> {
 }
 
 impl<'w> EntityRef<'w> {
+    /// # Safety
+    ///
+    /// Entity and location _must_ be valid for the provided world.
```

Comment:
```suggestion
    ///  - `entity` must be valid for `world`: the generation should match that of the entity at the same index.
    ///  - `location` must be sourced from `world`'s `Entities` and must exactly match the location for `entity`
    ///
    ///  The above is trivially satisfied if `location` was sourced from `world.entities().get(entity)`.
```

I know this was copied from `EntityMut`'s current safety comment, but it helps to be a bit more thorough. The corresponding safety comments from calling this function likely also need to be updated.